sim/CMakeLists.txt

```diff
@@ -1,37 +1,44 @@
 cmake_minimum_required(VERSION 3.16)
+include(FetchContent)
+# show progress for when fetching FetchContent dependencies
+set(FETCHCONTENT_QUIET OFF)
+# use new policy for options
+set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
 project(bentobox)
 
-# Source files
+## Source files
 set(SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
-set(LIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/lib")
 set(ICS_DIR "${SRC_DIR}/ics")
 set(UTIL_DIR "${SRC_DIR}/util")
-
-set(SOURCES
+set(LIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/lib")
+set(SOURCES 
     "${SRC_DIR}/main.cpp"
     "src/componentManager.h"
 )
-
-set(TESTS
+set(TESTS 
     "${SRC_DIR}/main.test.cpp"
     "src/componentManager.test.cpp"
 )
 
-# Executable
+## Executable
+set(TARGETS ${PROJECT_NAME} "${PROJECT_NAME}_test")
+# target bento simulator's executables to c++ 20
 set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
 add_executable(${PROJECT_NAME} ${SOURCES})
 add_executable("${PROJECT_NAME}_test" ${TESTS})
-target_include_directories(${PROJECT_NAME} PRIVATE ${SRC_DIR})
-
-# Link libraries
+foreach(target IN ITEMS ${TARGETS})
+    target_include_directories(${target} PRIVATE ${SRC_DIR})
+endforeach()
```

Comment:
This may be a bit overkill for two targets.
Though, if we're going down the route of having a list of targets, I guess we would need to have variable names for each target:
```
set(TARGET_MAIN ${PROJECT_NAME})
set(TARGET_TEST "${PROJECT_NAME}_test")
```
Then we can subsequently use the variables in the linking of subdir modules below.

Alternatively, we can check if the target ends with `_test` and then link those differently.

Reply:
> This may be a bit overkill for two targets.
Not really, it was getting really error prone to remember to link both the targets as our dependency list grows longer.

Replaced targets with `TARGET_SIM` and `TARGET_MAIN` variables.